src/html.cr

```diff
@@ -34,42 +36,27 @@ module HTML
     end
   end
 
-  # Returns a string where some named and all numeric character references
+  # Returns a string where named and numeric character references
   # (e.g. &gt;, &#62;, &x3e;) in *string* are replaced with the corresponding
-  # unicode characters. Only these named entities are replaced:
-  # apos, amp, quot, gt, lt and nbsp.
+  # unicode characters.
   #
   # ```
   # HTML.unescape("Crystal &amp; You") # => "Crystal & You"
   # ```
   def self.unescape(string : String) : String
-    return string unless string.includes? '&'
-
-    string.gsub(/&(apos|amp|quot|gt|lt|nbsp|\#[0-9]+|\#[xX][0-9A-Fa-f]+);/) do |string, _match|
-      match = _match[1]
-      case match
-      when "apos" then "'"
-      when "amp"  then "&"
-      when "quot" then "\""
-      when "gt"   then ">"
-      when "lt"   then "<"
-      when "nbsp" then " "
-      when /\A#0*(\d+)\z/
-        n = $1.to_i
-        if n <= Char::MAX_CODEPOINT
-          n.unsafe_chr
-        else
-          "&##{$1};"
-        end
-      when /\A#x([0-9a-f]+)\z/i
-        n = $1.to_i(16)
-        if n <= Char::MAX_CODEPOINT
-          n.unsafe_chr
-        else
-          "&#x#{$1};"
-        end
+    string.gsub(/&(?:([a-zA-Z]+;?)|\#([0-9]+);?|\#[xX]([0-9A-Fa-f]+);?)/) do |string, match|
```

Comment:
You should remove the first `;?` and make it just `;` since it'll never match inside `HTML::*_ENTITIES` without the `;` anyway.

Reply:
There are entities without ';'

Reply:
Like for example &amp

Reply:
I apologise for attempting to apply logic to HTML.

Reply:
I have a secret: I had it implemented like you asked me too until I realized &amp and a few others existed :-P

Reply:
This regex won't work for an entity that is not delimited by `;` but immediately followed by a ascii letter. For example `&ampd` should return `&d`.
This is diffiecult to handle properly. It requires either a custom parse tree or multiple checks if one of those not-delimited character sequences is matched. There are 106 of them ranging from 3 to 7 charachters.

Reply:
Fixed